core/core_integrations/slack.js

```diff
@@ -17,596 +15,553 @@ inTransaction = {},
+            token: teamInfo.token,
+            channel: teamInfo.channel_id,
+            username: exports.config.name,
+            link_names: 1,
+            text: message,
+            unfurl_links: true,
+            icon_url: exports.config.icon
+        };
+        addMessageToQueue(teamInfo.team_id, body, 'https://slack.com/api/chat.postMessage');
+    }
+    else {
+        console.debug('slack-> No slack team found!!!');
+    }
 },
 
 createThreadId = function(channelId, teamId) {
-	return channelId + '\0' + teamId;
+    return channelId + '\0' + teamId;
 },
 
 sendFile = function(type, file, description, thread) {
-	var teamInfo = getChannelIdAndTeamId(thread);
-
-	if (teamInfo.token !== null) {
-		var body = {
-			"token": teamInfo.token,
-			"channel": teamInfo.channel_id,
-			"file": file,
-			"filetype": type,
-			"title": description
-		};
-		addMessageToQueue(teamInfo.team_id, body, 'https://slack.com/api/files.upload');
-	}
-	else {
-		console.debug('slack-> No slack team found!!!');
-	}
+    var teamInfo = getChannelIdAndTeamId(thread);
+
+    if (teamInfo.token !== null) {
+        var body = {
+            token: teamInfo.token,
+            channel: teamInfo.channel_id,
+            file: file,
+            filetype: type,
+            title: description
+        };
+        addMessageToQueue(teamInfo.team_id, body, 'https://slack.com/api/files.upload');
+    }
+    else {
+        console.debug('slack-> No slack team found!!!');
+    }
 },
 
 renameChannel = function(title, thread) {
-	var teamInfo = getChannelIdAndTeamId(thread);
-
-	if (teamInfo.token !== null) {
-		var body = {
-			"token": teamInfo.token,
-			"channel": teamInfo.channel_id,
-			"name": title
-		};
-		addMessageToQueue(teamInfo.team_id, body, 'https://slack.com/api/channels.rename');
-	}
-	else {
-		console.debug('slack-> No slack team found!!!');
-	}
+    var teamInfo = getChannelIdAndTeamId(thread);
+
+    if (teamInfo.token !== null) {
+        var body = {
+            token: teamInfo.token,
+            channel: teamInfo.channel_id,
+            name: title
+        };
+        addMessageToQueue(teamInfo.team_id, body, 'https://slack.com/api/channels.rename');
+    }
+    else {
+        console.debug('slack-> No slack team found!!!');
+    }
 },
 
-getNextMessageId = teamId => {
-	return exports.config.slack_teams[teamId].event_id++;
-}
+getNextMessageId = function(teamId) {
+    return teamData[teamId].event_id++;
+},
 
-sendTyping = function(thread) {
-	var slackTeams = exports.config.slack_teams,
-		teamInfo = getChannelIdAndTeamId(thread),
-		teamId = teamInfo.team_id,
-		body = {
-			'type':'typing',
-			'channel': teamInfo.channel_id
-		};
-
-	body.id = getNextMessageId(teamId);
-	body = JSON.stringify(body);
-
-	console.debug("slack-> send typing indication");
-	sendSocketMessage(teamId, body);
+sendSocketMessage = function(type, teamId, body) {
+    var socket = sockets[teamId];
+    if (socket !== null) {
+        switch(type) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Keyword "switch" must be followed by whitespace.](https://www.codacy.com/app/theprojectshop/Kassy/file/2571707054/issues/source?bid=3282987&fileBranchId=3288593#l136)